### PR TITLE
Fixes typescript and build issues

### PR DESCRIPTION
**Pull request by bretep**
### ✅ Checklist

- [x] `npx changeset` was attached.

No new features. Only fixes project setup and build when following the README

### 📝 Description
I ran into project setup issues in preparation for adding PulseChain support to ledger-live. These should be non-breaking changes.

- Fixed issues related to TypeScript and the build process in multiple files across the project.
- Enforce pnpm version otherwise, the lock file will change.
- The "target" field in tsconfig.json files is changed from "es5" to "ES2020", and changes have been made in other configuration files.
- The versions of some dependencies like "@solana/spl-token" and "@solana/web3.js" in package.json files are updated.

### Tests performed

1. Cloned branch to new location
2. Followed the README.md
Ran the following:
```
pnpm install
pnpm build:libs
pnpm build:tests
pnpm build:dummy-apps
pnpm build:dummy-wallet-app
pnpm build:dummy-live-app
pnpm build:actions
pnpm build:cli
pnpm build:coin
pnpm build:domain
pnpm build:network
pnpm build:config
pnpm build:llc
pnpm build:lld:deps
pnpm build:lld
pnpm build:llm:deps
pnpm build:ljs
pnpm build:web-tools
```

All were built without errors, which was not the case before this commit.
---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Comment by bretep** [comment]
I'm not able to see the failures. Please let me know if there is something I need to resolve.

File: package.json [hunk 1]
```diff
@@ -177,6 +177,10 @@
     "unimported": "1.30.0",
     "zx": "^7.2.2"
   },
+  "engines": {
+    "pnpm": "9.7.0"
```
Comment (Justkant) [inline package.json:181]: If you check the `.prototools` file, you'll see that we use `9.4.0` of `pnpm`
```suggestion
    "pnpm": "9.4.0"
```

In the past I advocated for the use of `packageManager` field which is better than engines IMO because engines is more intended for consumers of a lib not developers AFAIK.
Probably better to use `packageManager` instead of `engines` here

File: libs/ledger-live-common/package.json [hunk 2]
```diff
@@ -258,7 +258,7 @@
     "@ledgerhq/types-cryptoassets": "workspace:^",
     "@ledgerhq/types-devices": "workspace:^",
     "@ledgerhq/types-live": "workspace:^",
-    "@solana/web3.js": "1.77.3",
+    "@solana/web3.js": "1.95.2",
```
Comment (Justkant) [inline libs/ledger-live-common/package.json:261]: Not sure we want to update unrelated libs in the same PR that just touches typescript and build stuff

File: apps/ledger-live-mobile/ios/Podfile [hunk 3]
```diff
@@ -8,6 +8,7 @@ require Pod::Executable.execute_command('node', ['-p',
 
 
 platform :ios, '13.4'
+use_modular_headers!
```
Comment (Justkant) [inline apps/ledger-live-mobile/ios/Podfile:11]: We already setup `use_frameworks! :linkage => linkage.to_sym` below that will be static or dynamic depending on env vars
So I don't think you need to add anything here, as seen here too: https://stackoverflow.com/a/53265888

File: apps/ledger-live-mobile/android/app/build.gradle [hunk 4]
```diff
@@ -128,7 +128,7 @@ android {
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
         versionCode 36176158
-        versionName "3.40.1"
+        versionName "3.47.2"
```
Comment (Justkant) [inline apps/ledger-live-mobile/android/app/build.gradle:131]: Not sure you need to touch this

File: apps/ledger-live-mobile/ios/ledgerlivemobile/Info.plist [hunk 5]
```diff
@@ -35,7 +35,7 @@
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>3.40.1</string>
+	<string>3.47.2</string>
```
Comment (Justkant) [inline apps/ledger-live-mobile/ios/ledgerlivemobile/Info.plist:38]: Not sure you need to touch this either because it's handle by scripts during the deploy on main/prod

File: libs/coin-modules/coin-solana/package.json [hunk 6]
```diff
@@ -53,8 +53,8 @@
     "@ledgerhq/logs": "workspace:^",
     "@ledgerhq/types-cryptoassets": "workspace:^",
     "@ledgerhq/types-live": "workspace:^",
-    "@solana/spl-token": "^0.3.7",
-    "@solana/web3.js": "1.77.3",
+    "@solana/spl-token": "^0.4.8",
+    "@solana/web3.js": "1.95.2",
```
Comment (Justkant) [inline libs/coin-modules/coin-solana/package.json:57]: Same as the other not sure about bumpimg solana deps, is this related to some types changes ?

File: libs/coin-modules/coin-solana/src/api/chain/web3.ts [hunk 7]
```diff
@@ -334,25 +334,29 @@ export function buildStakeWithdrawInstructions({
...
+  // Create the split instruction with the rentExemptReserve
+  const splitIx = StakeProgram.split(
+    {
+      authorizedPubkey: new PublicKey(authorizedAccAddr),
+      lamports: amount,
+      stakePubkey: new PublicKey(stakeAccAddr),
+      splitStakePubkey: new PublicKey(splitStakeAccAddr),
+    },
+    rentExemptReserve,
+  ).instructions[1];
```
Comment (Justkant) [inline libs/coin-modules/coin-solana/src/api/chain/web3.ts:353]: Not sure why you change logic in a PR that should only touch types and build issues

File: libs/coin-modules/coin-solana/src/buildTransaction.ts [hunk 8]
```diff
@@ -46,18 +46,24 @@ export const buildTransactionWithAPI = async (
...
   }
-  return buildInstructionsForCommand(commandDescriptor.command);
+  return await buildInstructionsForCommand(commandDescriptor.command, api);
 }
 
-function buildInstructionsForCommand(command: Command): TransactionInstruction[] {
+async function buildInstructionsForCommand(
+  command: Command,
+  api: ChainAPI,
+): Promise<TransactionInstruction[]> {
```
Comment (Justkant) [inline libs/coin-modules/coin-solana/src/buildTransaction.ts:66]: And that change might even be breaking now that you return a promise

File: libs/live-countervalues/package.json [hunk 9]
```diff
@@ -21,20 +21,22 @@
...
-    "@ledgerhq/logs": "workspace:*",
+    "@ledgerhq/cryptoassets": "workspace:*",
+    "@ledgerhq/live-env": "workspace:*",
     "@ledgerhq/live-network": "workspace:*",
     "@ledgerhq/live-promise": "workspace:*",
-    "@ledgerhq/live-env": "workspace:*"
+    "@ledgerhq/logs": "workspace:*",
+    "@ledgerhq/types-cryptoassets": "workspace:*",
+    "@ledgerhq/types-live": "workspace:*",
+    "url": "^0.11.4"
```
Comment (Justkant) [inline libs/live-countervalues/package.json:32]: Do we need to add url here ?

File: tools/actions/turbo-affected/build/main.js
Comment (Justkant) [inline tools/actions/turbo-affected/build/main.js:1]: We probably want to revert this change if it was only formatting changes, unless you did a rebuild of this script for a reason

File: apps/cli/src/commands/device/managerListApps.ts [hunk 10]
```diff
@@ -4,6 +4,11 @@ import { withDevice } from "@ledgerhq/live-common/hw/deviceAccess";
 import getDeviceInfo from "@ledgerhq/live-common/hw/getDeviceInfo";
 import { listAppsUseCase } from "@ledgerhq/live-common/device/use-cases/listAppsUseCase";
 import { deviceOpt } from "../../scan";
+import { ListAppsEvent, ListAppsResult } from "@ledgerhq/live-common/apps/types";
+
+function isResultEvent(event: ListAppsEvent): event is { type: "result"; result: ListAppsResult } {
+  return event.type === "result";
+}
```
Comment (Justkant) [inline apps/cli/src/commands/device/managerListApps.ts:11]: Nice, another small improvement we could do would be to create a type for `{ type: "result"; result: ListAppsResult }` and use it both here and in `ListAppsEvent`

**Review by Justkant — COMMENTED** [review]
Hello and thank you for the contribution

We'll need to have reviews from people maintaining our mono-repo and build system too

Could you also list/explain the issues you fixed with the build process in multiple files across the project ? I don't really see a lot of changes on the build side (unless it's the main script rebuild)

Could you also explain why you need to target `ES2020` in the tsconfig ? Also I don't think you modified the correct config as we currently share a single config for all libs in the monorepo and cli or web-tools are not the ones

About the pnpm version we already enforce a version with Proto and it's documented in the README

I also don't see a big reason to update dependencies of solana (unless big issues with types) and modifying the logic while we should only be fixing typescript/build issues

**Comment by bretep** [comment]
I will review and document each error and the reason for the package upgrades. 

By chance, have you checked out the repo on a clean computer, without global packages installed, and followed the readme for setup?

**Comment by bretep** [comment]
I took a step back because of your comment about `proto` to check and see why I would have received all these issues. It turns out that `proto` was not being used when working in my IDE, and I was using newer node versions.

When running outside of my IDE, I could clone and run through the README with no issues. Many of the errors that were stopping the build are now `warnings`.

Thank you for reviewing the PR and I apologize for the wasted time.
